packages/web/src/common/store/tipping/sagas.ts

```diff
@@ -263,7 +267,171 @@ function* confirmTipIndexed({
   return false
 }
 
+function* wormholeAudioIfNecessary({ amount }: { amount: number }) {
+  const walletClient = yield* getContext('walletClient')
+
+  const waudioBalanceWei = yield* call([
+    walletClient,
+    'getCurrentWAudioBalance'
+  ])
+  const audioWeiAmount = new BN(AUDIO(amount).value.toString()) as BNWei
+
+  if (isNullOrUndefined(waudioBalanceWei)) {
+    throw new Error('Failed to retrieve current wAudio balance')
+  }
+
+  // If transferring spl wrapped audio and there are insufficent funds with only the
+  // user bank balance, transfer all eth AUDIO to spl wrapped audio
+  if (audioWeiAmount.gt(waudioBalanceWei)) {
+    console.info('Converting Ethereum AUDIO to Solana wAUDIO...')
+
+    // Wait for a second before showing the notice that this might take a while
+    const showConvertingMessage = yield* fork(function* () {
+      yield delay(1000)
+      yield put(convert())
+    })
+    yield call([walletClient, walletClient.transferTokensFromEthToSol])
+    // Cancel showing the notice if the conversion was magically super quick
+    yield cancel(showConvertingMessage)
+  }
+}
+
 function* sendTipAsync() {
+  const audiusSdk = yield* getContext('audiusSdk')
+  const sdk = yield* call(audiusSdk)
+  const isNativeMobile = yield* getContext('isNativeMobile')
+
+  const sender = yield* select(getAccountUser)
+  const {
+    trackId,
+    user: receiver,
+    amount: stringAudioAmount,
+    onSuccessConfirmedActions,
+    source
+  } = yield* select(getSendTipData)
+
+  if (!sender || !receiver) {
+    return
```

Comment:
Wondering how we would end up in this state and whether it should generate an error?
And/or would there have been any status elsewhere that was set before calling this that we are leaving hanging?